src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        String[][] words = new String[data.length][4];
+
+        for (int i = 0; i < data.length; i++) {
+            words[i] = data[i].split(" ");
+        }
+
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append("\n");
+
+        for (int i = 0; i < names.length; i++) {
+            int totalSalary = 0;
+
+            for (int j = 0; j < words.length; j++) {
+                if (names[i].equals(words[j][1]) && isDateInRange(words[j][0], dateFrom, dateTo)) {
+                    int hours = Integer.parseInt(words[j][2]);
+                    int salaryPerHour = Integer.parseInt(words[j][3]);
+                    int salary = hours * salaryPerHour;
+                    totalSalary += salary;
+                }
+            }
```

Comment:
use one-dimensional array instead, it's simpler
something like that:
```suggestion
        for (String name : names) {
            int totalSalary = 0;
             
            for (int i = 0; i < data.length; i++) {
                String[] splittedData = words[i] = data[i].split(" ");
                if (name.equals(splittedData[1]) && isDateInRange(splittedData[0], dateFrom, dateTo)) {
                    int hours = Integer.parseInt(splittedData[2]);
                    int salaryPerHour = Integer.parseInt(splittedData[3]);
                    int salary = hours * salaryPerHour;
                    totalSalary += salary;
                }
            }
```
After refactoring please extract magic indexes to constants with meaningful naming